plugins/lsp/conform-nvim.nix

```
@@ -36,97 +86,159 @@ in
             };
         ```
       '';
-    };
 
-    formatOnSave = helpers.defaultNullOpts.mkNullable' {
-      type =
-        with helpers.nixvimTypes;
-        either strLuaFn (submodule {
-          options = {
-            lspFallback = mkOption {
-              type = types.bool;
-              default = true;
-              description = "See :help conform.format for details.";
-            };
-            timeoutMs = mkOption {
-              type = types.int;
-              default = 500;
-              description = "See :help conform.format for details.";
-            };
-          };
-        });
-      # Unknown plugin default
-      description = ''
-        If this is set, Conform will run the formatter on save.
-        It will pass the table to conform.format().
-        This can also be a function that returns the table.
-        See :help conform.format for details.
+      format_on_save =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            either strLuaFn (submodule {
+              freeformType = types.attrsOf anything;
+              options = {
+                inherit
+                  lsp_format
+                  timeout_ms
+                  quiet
+                  stop_after_first
+                  ;
+              };
+            })
+          )
+          { }
+          ''
+            If this is set, Conform will run the formatter on save.
+            It will pass the table to `conform.format()`.
+            This can also be a function that returns the table.
+            See `:help conform.format` for details.
+          '';
+
+      default_format_opts =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            either strLuaFn (submodule {
+              freeformType = types.attrsOf anything;
+              options = {
+                inherit
+                  lsp_format
+                  timeout_ms
+                  quiet
+                  stop_after_first
+                  ;
+              };
+            })
+          )
+          { }
+          ''
+            The default options to use when calling `conform.format()`.
+          '';
+
+      format_after_save =
+        defaultNullOpts.mkNullable
+          (
+            with types;
+            either strLuaFn (submodule {
+              freeformType = attrsOf anything;
+              options = {
+                inherit
+                  lsp_format
+                  timeout_ms
+                  quiet
+                  stop_after_first
+                  ;
+              };
+            })
+          )
+          { }
+          ''
+            If this is set, Conform will run the formatter asynchronously after save.
+            It will pass the table to `conform.format()`.
+            This can also be a function that returns the table.
+            See `:help conform.format` for details.
```

Personally I like to start a new paragraph for footnotes like _"See XYZ for more info."_

It may also be helpful to have either a line break (`\` before eol) or a paragraph break (a blankline) for somewhat separate thoughts such as _"This can also be a function that returns the table."_, though I feel less strongly about that :sweat_smile: 

```suggestion
            If this is set, Conform will run the formatter asynchronously after save.

            It will pass the table to `conform.format()`.
            This can also be a function that returns the table.

            See `:help conform.format` for details.
```

---

I'm also not keen on the over-use of _"it"_; I think it adds more clarity to be a bit more verbose about what _"it"_ actually is.

Ironically, I think my point is proven by the fact I don't know what _"it"_ should be replaced with, therefore I'll forgive you if you also aren't sure :joy: